### ISSUE TITLE
Fix Explicit Column Select Behaviour

### ISSUE THREAD

**Issue by eimihar** [envms/fluentpdo#105]
i found out that by default, i can't actually build a query with only a specific of column selection. because it appears they already selected all of the columns for the chosen table.

for example :

```
$query = $db->from('article')->select('articleId');

echo $query->getQuery(); // will return SELECT article.*, articleId FROM article
// i expect SELECT articleId FROM article
```

is this normal.? or is it a feature or bug?

thank you.

**Comment by motin** [comment]
I wonder this as well. I only want specific columns returned from the base table, but tablename.* gets inserted into the query. 

Suggestion: Allow starting with `$fpdo->select('attribute1', 'attribute2')` before executing `->from('tablename')`

In that case, only include `tablename.attribute1, tablename.attribute2` instead of `tablename.*`

**Comment by rorc** [comment]
$db->from('article')->select(null)->select('articleId');

**Comment by lichtner** [comment]
because usually you need `table.*` only rarely something else,
but also this is most usual problem for many people 
so may be it could be configurable...

**Comment by jkufner** [comment]
Are you sure? I see `->select(null)` very often. The default should be
"undefined", requiring `->select('*')` to be explicitly stated.

**Comment by TCB13** [comment]
I agree with @motin suggestion. It's more natural and close to the real SQL.

However, after implementing the possibility to define first the `select` attributes we should also require `select` to be always explicitly stated as @jkufner said.

**Comment by rorc** [comment]
Need add other joins...
$fpdo->from('table')->leftOuterJoin();
$fpdo->from('table')->rightOuterJoin();
etc...

**Comment by lichtner** [comment]
ok guys, I know, it was stupid decision add this default select for all select.

It is not problem to change this behavior, it is very easy, but I don't know how to do it without BC break or with minimum pain for all fpdo users. Because if you expect:

    $fpdo->from('article')->select('user.name')

select `article.*, user.name` and after this change query select only `user.name`. 
I expect all application based on fpdo will be broken and many developers will curse me ;-)

If I release 2.0.0 with this change it means for you all apps rewrite almost every query like this:

    $fpdo->from('table')
    // rewrite
    $fpdo->from('table')->select('table.*')
  
Do you still believe it worth it, guys? ;-)
Or somebody have some better solution?
And who prepare pull request with all updated tests? ;-)

**Comment by TCB13** [comment]
No rewrite needed. FluentPDO should be smart enough to do a `SELECT *` on `$fpdo->from('table')` and a `SELECT name` on `$fpdo->from('article')->select('name')`.

**Comment by lichtner** [comment]
@TCB13  it is no problem add `SELECT *` if no select was added, but problem is if some select was already added. And now is question: When I should add also `SELECT *`? May be I don't understand what do you think....  try to send pull request...

**Comment by lichtner** [comment]
I made this: https://github.com/lichtner/fluentpdo/pull/112
As you can see many tests failed. Anyone has better solution?

**Comment by rorc** [comment]
Just add settings on init fpdo, for example $pdo = new FPDO(); $pdo->disableAutomaticSelectFromFirstTable()
=)

**Comment by PawelDecowski** [comment]
Just change it in 2.0. Major version increment is exactly for that — introducing BC changes.

For 1.*  I’d suggest adding an option in the constructor to disable selecting all fields when `select` is called.

**Comment by cbornhoft** [comment]
Closing, noted for future feature request.

**Comment by cbornhoft** [comment]
Fixed in 2.0
